### ISSUE TITLE
Consider using `git restore` and `git switch`

### ISSUE THREAD

**Issue by bast** [coderefinery/git-intro#326]
> `git restore` and `git switch` were added in git 2.23 (released
2019-08-16).  Sometime much later (2021, 2022, or 2023) perhaps, the
older alternatives can be demoted or removed.

**Comment by rkdarst** [comment]
tldr: what about after 3 years (autumn this year)?

I guess you know I'm very conservative and biased towards "default works for everyone" (not just in the lesson but most other computers one might use - I assume moving to your cluster and "git doesn't work anymore" is the worst thing that can happen).

What about late this year, after there have been two Ubuntu LTS releases and 3 years overall since the release?  Anyone know git versions on mac - well, our instructions say to install it ourselves.  Is it also there by default?
I guess windows users install it themselves, so we are probably safe there.
What about our clusters?  Our cluster's default OS git is 1.8 - a module is available but that's yet another complexity to learn.

**Comment by bast** [comment]
+1 to moving to this late this year or even early 2023.

I have introduced the modern commands in https://coderefinery.github.io/git-intro/recovering/#undoing-your-recent-uncommitted-and-unstaged-changes-preserves-history

However:
- there is a note for older Git
- this is an optional episode which I will not teach during workshops but only refer people to as optional reference

**Comment by rkdarst** [comment]
shall we check what OSs support it by default, if we don't expect people to install new git if they already have it?
ubuntu 20.04 and greater
debian 10 from 2019 and greater
windows and mac CodeRefinery instructions imply people should manually install - does everyone do that?

I guess most recent mainstream people will have new enough now, so I can find less and less reason to wait.  But I still have dreams of users going to some other system and what they learned doesn't work... I know I use plenty that would fit that category...

**Comment by bast** [comment]
Maybe adding a box/section "if you are on older git or on some cluster and things fail"?

**Comment by plaihonen** [comment]
In the lesson material, https://coderefinery.github.io/git-intro/archaeology/#git-checkout-b-to-inspect-code-in-the-past.

It would be good to indicate from which git version this "switch" functionality has been added.
E.g. "On newer Git versions (2.23 and above) this is the preferred command:"
